### PR TITLE
Minor Sentence Structure Reworks and Adding Sources

### PR DESCRIPTION
**Pull request by jskripchuk**
## Proposed changes

- Reworked sentence structure for long sentences that were confusing.
- Added sources for
  - US Missile Bug: https://nsarchive2.gwu.edu/nukevault/ebb371
    - One incident that I like with a similar gravitas is the 1983 Soviet False alarm. It has the added bonus of being easily explainable to audiences as 'the computer mistook a rare alignment of high altitude clouds and the satellite orbit as a nuclear missile' (https://en.wikipedia.org/wiki/1983_Soviet_nuclear_false_alarm_incident)
  - Software Crisis: https://en.wikipedia.org/wiki/Software_crisis

### Checklist

*Change the space in the box to an `x` for those you have completed.
You can also fill these out after creating the pull request.
If you're unsure about any of them, don't hesitate to ask.
We're here to help!
This is simply a reminder of what we are going to look for before merging your change.*

- [x] I have read the contribution guidelines.
- [x] I have linked any relevant existing issues/suggestions in the description above (include `#???` in your description to reference an issue, where `???` is the issue number).
- [x] The pull request is requesting a merge into the `develop` branch.
- [x] I have added necessary documentation (if appropriate).
- [x] If I've added/modified/deleted a third-party system, I've updated the relevant license files.

### CODE REVIEW

File: csfieldguide/chapters/content/en/software-engineering/software-engineering.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # Challenges in teaching Software Engineering
 
 Software engineering is a challenging topic to teach for several reasons.
-One is that many, if not most, textbooks aren't written by practising software engineers, and often contain outdated ideas that are difficult for students to understand anyway (for example, the waterfall model is often taught near the start, and yet often doesn’t work in practice, and in fact was originally written to tell people how *not* to do software engineering).
+One reason is that many, if not most, textbooks aren't written by practising software engineers, and often contain outdated ideas that are difficult for students to understand. For example, the waterfall model is often taught near the start, and yet often doesn’t work in practice, and in fact was originally written to tell people how *not* to do software engineering.
```
Comment (eAlasdair) [inline csfieldguide/chapters/content/en/software-engineering/software-engineering.md]: Regarding project style, we start new sentences on a new line &mdash; It makes it easier to understand when we revisit/compare content, and the single newline is replaced with a space when the document is viewed

Comment (eAlasdair) [inline csfieldguide/chapters/content/en/software-engineering/software-engineering.md]: Regarding project style, we start new sentences on a new line — It makes it easier to understand when we revisit/compare content, and the single newline is replaced with a space when the document is viewed

Comment (jskripchuk) [reply csfieldguide/chapters/content/en/software-engineering/software-engineering.md]: Smart idea. Updated.

**Comment by jskripchuk** [comment]
> Excellent, do you have the permissions to merge it or do you need us to?

Nope, so you all are going to have to merge.

**Pull request merged**